### PR TITLE
Users can run actions if they have access to the database

### LINKED ISSUE #28425: Non-admin users can't run actions
**Describe the bug**
Non-admin users with full permissions can't run actions

**To Reproduce**
1. Run an instance with a DB with actions
2. Create a dashboard with any valid action
3. Log in as a non-admin user with full permissions
4. Open the dashboard and try to run any action
5. Backend replies with 403 "You don't have permissions to do that."

**Expected behavior**
Non-admin users should be able to run actions

**Information about your Metabase Installation:**

`master` branch

### PR DESCRIPTION
**Pull request by tsmacdonald**
[Fixes #28425]

Takes care of the "Can run?" section of this table:

image

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29088)

### CODE REVIEW

**Review by kulyk — APPROVED** [review]
Works great, thank you! @tsmacdonald I think we need to add the "backport" label here?

**Review by qnkhuat — CHANGES_REQUESTED** [review]
@tsmacdonald or @kulyk I followed the reproduction steps in the original issue and successfully execute the action as Rasta user. Could you double-check if the steps are correct?

Also I'm not sure the fix makes sense, we're changing from `/db/1/` to `/db/1/schema`, isn't it supposed to be more restrictive than before?

**Comment by tsmacdonald** [comment]
@qnkhuat just pushed up a new commit since apparently we're supposed to be *very* permissive. Here's the truth table that Bruno signed off on. This PR is just for the execution part. Action creation is covered by #29171 

image

File: test/metabase/api/dashboard_test.clj [hunk 1]
```diff
@@ -2938,52 +2946,52 @@
                               (mt/user-http-request :crowberto :post 400 execute-path {:parameters {"id" 1 "name" "Blueberries" "price" 1234}})))))))))))
 
 ;; Not relevant for 46; uncomment and make work for 47
```
Comment (tsmacdonald) [inline test/metabase/api/dashboard_test.clj:2948]:
```suggestion
```

**Review by qnkhuat — DISMISSED** [review]
I think this looks great, the code and permissions table checks out.

But sorry to be the bad guy to request changes here but since we're revisiting action perms, I think we should add some low-level tests in `test/metabase/query_processor/middleware/permissions_test.clj` for `check-query-action-permissions*` too

**Review by calherries — APPROVED** [review]
Looks good

File: test/metabase/query_processor/middleware/permissions_test.clj [hunk 2]
```diff
@@ -211,6 +211,31 @@
...
+          (mt/with-temp Collection [collection]
+            (mt/with-temp Card [{model-id :id} {:collection_id (u/the-id collection)
+                                                :dataset_query query}]
+              (testing "are granted by default"
+                (check! query))
+              (testing "are revoked without access to the model"
+                (binding [qp.perms/*card-id* model-id]
+                  (is (thrown-with-msg?
+                       ExceptionInfo
+                       (re-pattern #"You do not have permissions to view Card [\d,]+")
```
Comment (metamben) [inline test/metabase/query_processor/middleware/permissions_test.clj:230]:
```suggestion
                       #"You do not have permissions to view Card [\d,]+"
```

**Review by metamben — APPROVED** [review]
LGTM

**Pull request merged**